plugin/core/types.py

```diff
@@ -447,6 +451,34 @@ def matches(self, view: sublime.View) -> bool:
         return any(f(view) for f in self.filters) if self.filters else True
 
 
+def match_file_operation_filters(
+    file_operation_options: FileOperationRegistrationOptions, path: str, view: sublime.View | None
+) -> bool:
+    def matches(file_operation_filter: FileOperationFilter) -> bool:
+        pattern = file_operation_filter.get('pattern')
+        scheme = file_operation_filter.get('scheme')
+        if scheme and view:
+            uri = view.settings().get("lsp_uri")
+            if isinstance(uri, str) and parse_uri(uri)[0] != scheme:
+                return False
+        if pattern:
+            matches = pattern.get('matches')
+            if matches == FileOperationPatternKind.File and os.path.isdir(path):
+                return False
+            if matches == FileOperationPatternKind.Folder and os.path.isfile(path):
+                return False
+            options = pattern.get('options', {})
+            flags = GLOBSTAR | BRACE
+            if options.get('ignoreCase', False):
+                flags |= IGNORECASE
+            if not globmatch(path, pattern['glob'], flags=flags):
+                return False
+        return True
+
+    filters = file_operation_options.get('filters')
+    return any(matches(_filter) for _filter in filters) if filters else True
```

Comment:
Here also filters are never undefined so can get using `file_operation_options['filters']`.

As far as filter list being empty, do we know for sure that in that case it should be a match?